std/typecons.d

```diff
@@ -3431,6 +3431,36 @@ Params:
     assert(!npi.isNull);
 }
 
+	/**
+      If they are both null, then they are equal. If one is null and the other
+      is not, then they are not equal. If they are both non-null, then they are
+      equal if their values are equal.
+      */
+    bool opEquals()(auto ref const(typeof(this)) rhs) const
+    {
+        if (isNull)
+            return rhs.isNull;
+        if (rhs.isNull)
+            return false;
+        return _value == rhs._value;
+    }
```

Comment:
This can be simplified to return `_value == rhs._value`

Reply:
And by that I mean, the whole function can reduce to that :) I realize reading my comment may not be clear.